versionist.conf.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const execSync = require('child_process').execSync;
+const yaml = require('yaml');
+const fs = require('fs');
+const regex = /(docker:\/\/ghcr.io\/balena-io\/deploy-to-balena-action):(.*)/i;
+
+const getAuthor = (commitHash) => {
+  return execSync(`git show --quiet --format="%an" ${commitHash}`, {
+    encoding: 'utf8'
+  }).replace('\n', '');
+};
+
+const isIncrementalCommit = (changeType) => {
+  return Boolean(changeType) && changeType.trim().toLowerCase() !== 'none';
+};
+
+module.exports = {
+  // This setup allows the editing and parsing of footer tags to get version and type information,
+  // as well as ensuring tags of the type 'v<major>.<minor>.<patch>' are used.
+  // It increments in a semver compatible fashion and allows the updating of NPM package info.
+  editChangelog: true,
+  parseFooterTags: true,
+  getGitReferenceFromVersion: 'v-prefix',
+  incrementVersion: 'semver',
+  updateVersion: (cwd, version, callback) => {
+    const packageFile = `${cwd}/package.json`;
+    const packageData = require(packageFile);
+    packageData.version = version;
+    fs.writeFileSync(packageFile, JSON.stringify(packageData, null, 2));
+
+    const actionYml = `${cwd}/action.yml`;
+
+    function replacer(match, p1, p2, offset, string, groups) {
+      return [p1, `v${version}`].join(':')
+    }
+
+    const action = yaml.parse(fs.readFileSync(actionYml, 'utf8'));
+    action.runs.image = action.runs.image.replace(regex, replacer);
```

Comment:
You can do this with a string replacement rather than requiring a function
```suggestion
    const action = yaml.parse(fs.readFileSync(actionYml, 'utf8'));
    action.runs.image = action.runs.image.replace(regex, `$1:v${version}`);
```
where `$1` is the first matched group